react-stopwatch-component/src/stopwatch.jsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+
+class StopWatch extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.state = { time: 0, paused: true };
+    this.buttonClick = this.buttonClick.bind(this);
+    this.faceClick = this.faceClick.bind(this);
+  }
+
+  componentDidUpdate() {
+    if (!this.intID && this.state.paused === false) {
+      this.intID = setInterval(() =>
+        this.tick(), 1000
+      );
+      console.log('setInt ran', this.state.time);
+    } else if (this.intID && this.state.paused === true) {
+      clearInterval(this.intID);
+      this.intID = null;
+      console.log('clear ran');
+    }
+  }
+
+  tick() {
+    let newTime = this.state.time + 1;
+    console.log(newTime);
+    this.setState(prevState => (
+      { time: newTime }
+    ));
```

Comment:
Using the callback function for `setState` here is overkill, it is better to just pass `setState` an object:

```JSX
this.setState({ time: newTime });
```

Reply:
fixed!